gpu/octree/src/cuda/octree_host.cu

```diff
@@ -145,10 +145,12 @@ namespace
 }
 
 void pcl::device::OctreeImpl::radiusSearchHost(const PointType& query, const float radius, std::vector<int>& out, std::vector<float>& sqr_distances, const int max_nn, const bool compute_all_distances) const
-{            
-    out.clear();  
+{
+    out.reserve(max_nn);
+    sqr_distances.reserve(max_nn);
+    out.clear();
```

Comment:
I'm more used to seeing
1. call clear
2. call reserve

but yo made me read about it and now I discovered clear doesn't touch capacity, so it should make a difference.

You didn't clear the content of `sqr_distances` though.